ocf_data_sampler/config/model.py

```diff
@@ -208,27 +199,10 @@ def items(self):
         return self.root.items()
 
 
-# noinspection PyMethodParameters
-class GSP(DataSourceMixin, TimeResolutionMixin, DropoutMixin):
+class GSP(TimeWindowMixin, DropoutMixin):
```

Comment:
Should this be a DataSourceBase?

Reply:
GSP(DataSourceMixin), does that work? Or perhaps there is a reason for it to be like it is. I may have missed something

Reply:
I see what you mean - GFS doesn't have crop size parameters, which are required in DataSourceMixin. I wouldn't want to make them optional, and since GFS only needs zarr path from there, I thought this was better than dissolving DataSourceMixin/leaving only crop parameters in it as that means zarr path will have to be added individually to every source. But this might not be the best solution either, I'm open to suggestions here.

Reply:
To be honest I'm not wild about how DataSourceMixin is grouped right now - it has both zarr path and spacial cropping which aren't really related, but we have to add zarr path to everything so I wanted to put it into a mixin, and a mixin for zarr path alone would increase unnecessary code. But this is clearly not a universal solution so maybe I shouldn't have done that

Reply:
I see `DataSourceMixin` has some spacial element to it too. Perhaps just leave it how it is for the moment with `TimeWindowMixin` and `DropoutMixin`. 

You could have `ImageDataSourceMixin` which inherits from `DataSourceMixin` and has spacial cropping elements. Just an idea, 

Potentially it might be simplier leaving it how it is right now in the PR

Reply:
Yeah good point, done that now, thanks! I've added `zarr_path` back into each config for now, and I think that's for the best. We can add another mixin just for that but I don't think there's a point in doing it, and also this way it's clearer which parameters each one has without looking up mixins.